internal/core/string.rs

```diff
@@ -212,6 +213,18 @@ impl From<&String> for SharedString {
     }
 }
 
+impl From<SmolStr> for SharedString {
```

Comment:
I'd like to avoid this dependency in the runtime library. 
Even if it was optional, it would still be exposed in the poublic interface making it impossible to upgrade the smol_str dependency if it changes major version.
Would it be terrible to remove these two impl?

Reply:
done, requires some more `foo.as_str().into()` but otherwise is not a big deal.